packages/flame_tiled/test/tiled_test.dart

```diff
@@ -463,6 +463,128 @@ void main() {
       expect(pngData, matchesGoldenFile('goldens/pointy_hex_odd.png'));
     });
   });
+
+  group('isometric staggered', () {
+    late Uint8List pngData;
+    late TiledComponent component;
+
+    Future<TiledComponent> setupMap(
+      String tmxFile,
+      String imageFile,
+      Vector2 destTileSize,
+    ) async {
+      Flame.bundle = TestAssetBundle(
+        imageNames: [
+          imageFile,
+        ],
+        mapPath: 'test/assets/$tmxFile',
+      );
+      return component = await TiledComponent.load(
+        tmxFile,
+        destTileSize,
+      );
+    }
+
+    test('x + odd', () async {
+      await setupMap(
+        'iso_staggered_overlap_x_odd.tmx',
+        'dirt_atlas.png',
+        Vector2(128, 64),
+      );
+
+      expect(component.size, Vector2(320, 288));
+
+      final canvasRecorder = PictureRecorder();
+      final canvas = Canvas(canvasRecorder);
+      component.tileMap.render(canvas);
+      final picture = canvasRecorder.endRecording();
+
+      final image = await picture.toImageSafe(320, 288);
+      pngData = (await image.toByteData(format: ImageByteFormat.png))!
+          .buffer
+          .asUint8List();
+
+      expect(
+        pngData,
+        matchesGoldenFile('goldens/iso_staggered_overlap_x_odd.png'),
+      );
+    });
+
+    test('x + even + half sized', () async {
+      await setupMap(
+        'iso_staggered_overlap_x_even.tmx',
+        'dirt_atlas.png',
+        Vector2(128 / 2, 64 / 2),
+      );
+
+      expect(component.size, Vector2(320 / 2, 288 / 2));
+
+      final canvasRecorder = PictureRecorder();
+      final canvas = Canvas(canvasRecorder);
+      component.tileMap.render(canvas);
+      final picture = canvasRecorder.endRecording();
+
+      final image = await picture.toImageSafe(160, 144);
+      pngData = (await image.toByteData(format: ImageByteFormat.png))!
+          .buffer
+          .asUint8List();
+
+      expect(
+        pngData,
+        matchesGoldenFile('goldens/iso_staggered_overlap_x_even.png'),
+      );
+    });
+
+    test('y + odd + half', () async {
+      await setupMap(
+        'iso_staggered_overlap_y_odd.tmx',
+        'dirt_atlas.png',
+        Vector2(128 / 2, 64 / 2),
+      );
+
+      expect(component.size, Vector2(576 / 2, 160 / 2));
+
+      final canvasRecorder = PictureRecorder();
```

Comment:
Nit: Maybe you could use Flame layers for a cleaner code?

Reply:
Hi! I'd love to have cleaner code - is this what you're referring to?

https://pub.dev/documentation/flame/latest/layers/Layer-class.html

And is there any examples in the flame test code I can use to get up to speed on it?

Reply:
Maybe this could help: https://examples.flame-engine.org/#/Rendering_Layers
Get to the code by pressing `< >` in the upper right corner.

Reply:
The question that I think remains: is this about the test code or is this a suggestion that each RenderableTileLayer is now a FlameLayer?

Reply:
(a search for `PreRenderedLayer` and `DynamicLayer` in *test.dart is not turning up anything).

Reply:
I was thinking on the tests itself, indeed PreRenderLayer and DunamicLayer isn't used in any tests. But since those classes basically wrap a PictureRecoder, it could make these tests cleaner, since you won't need to instantiate the Canvas and the PictureRecorder yourself.

But again this is just a nit. PR looks good to me already.

Reply:
I think it is several years since anyone did any real changes to that file, so from before we required tests unfortunately.

Reply:
Yep - I could do a non-lazy thing and move the canvas/picturerecorder/give me an image code into a function since its the same method over and over.. I can do this later tonight (lunch break over):

```dart
Future<Uint8List> generatePngData(Component component, Vector2 size) async {
      final canvasRecorder = PictureRecorder();
      final canvas = Canvas(canvasRecorder);
      component.tileMap.render(canvas);
      final picture = canvasRecorder.endRecording();
      
      final image = await picture.toImageSafe(size.x, size.y);
      return (await image.toByteData(format: ImageByteFormat.png))!
          .buffer
          .asUint8List();      
}
```

Reply:
No rush, love that Flame is getting some love even on the lunch breaks. :smile:

Reply:
done